### PR TITLE
Add Computation Expression Benchmarks

### PR DESCRIPTION
**Pull request by dawedawe**
## Description

While working on improving the compilation speed for CEs, it might be a good idea to have some benchmarks in the repo with different CE like with/without Custom Ops, different levels of nesting, different length.
I disabled CE100xnest10.fs for now, because it really runs for a long time...

|                     Method |             Mean |           Error |          StdDev |        Gen0 |       Gen1 |       Gen2 |      Allocated |
|--------------------------- |-----------------:|----------------:|----------------:|------------:|-----------:|-----------:|---------------:|
|        Check100xNestingOf1 |         929.3 us |        11.66 us |        10.34 us |      1.9531 |          - |          - |       397.2 KB |
|        Check100xNestingOf5 |         965.3 us |        11.38 us |        10.09 us |      3.9063 |          - |          - |      467.41 KB |
|        Check200xNestingOf5 |       1,118.4 us |        12.73 us |        11.91 us |      3.9063 |     1.9531 |     1.9531 |      544.12 KB |
|       Check100xNestingOf10 |       2,776.8 us |        57.31 us |       165.35 us |           - |          - |          - |      610.59 KB |
|   CheckNesting1WithCustOps |       1,037.0 us |        14.78 us |        13.83 us |      3.9063 |          - |          - |      486.02 KB |
|   CheckNesting5WithCustOps |       1,353.8 us |        26.11 us |        31.08 us |      7.8125 |     5.8594 |     3.9063 |      703.79 KB |
|      Compile100xNestingOf1 |     611,959.0 us |     7,753.45 us |     6,474.48 us |   1000.0000 |          - |          - |   249178.91 KB |
|      Compile100xNestingOf5 |   6,232,674.9 us |    40,350.93 us |    37,744.28 us |  18000.0000 |  4000.0000 |  2000.0000 |  2211470.08 KB |
|      Compile200xNestingOf5 |   6,220,882.7 us |    39,973.98 us |    35,435.89 us |  19000.0000 |  4000.0000 |  3000.0000 |   2208148.9 KB |
|     Compile100xNestingOf10 | 113,578,524.0 us | 2,248,102.35 us | 6,154,143.90 us | 415000.0000 | 37000.0000 | 23000.0000 | 53534013.59 KB |
| CompileNesting1WithCustOps |       1,088.9 us |        20.22 us |        39.91 us |      3.9063 |          - |          - |      484.16 KB |
| CompileNesting5WithCustOps |       1,352.9 us |        25.09 us |        24.65 us |      7.8125 |     3.9063 |     3.9063 |      703.77 KB |

## Checklist

- [ ] Test cases added
- [ ] Performance benchmarks added in case of performance changes
- [ ] Release notes entry updated:
    > Please make sure to add an entry with short succinct description of the change as well as link to this pull request to the respective release notes file, if applicable.
    >
    > Release notes files:
    > - If anything under `src/Compiler` has been changed, please make sure to make an entry in `docs/release-notes/.FSharp.Compiler.Service/<version>.md`, where `<version>` is usually "highest" one, e.g. `42.8.200`
    > - If language feature was added (i.e. `LanguageFeatures.fsi` was changed), please add it to `docs/releae-notes/.Language/preview.md`
    > - If a change to `FSharp.Core` was made, please make sure to edit `docs/release-notes/.FSharp.Core/<version>.md` where version is "highest" one, e.g. `8.0.200`.

    > Information about the release notes entries format can be found in the [documentation](https://fsharp.github.io/fsharp-compiler-docs/release-notes/About.html).
    > Example:
    > * More inlines for Result module. ([PR #16106](https://github.com/dotnet/fsharp/pull/16106))
    > * Correctly handle assembly imports with public key token of 0 length. ([Issue #16359](https://github.com/dotnet/fsharp/issues/16359), [PR #16363](https://github.com/dotnet/fsharp/pull/16363))
    > *`while!` ([Language suggestion #1038](https://github.com/fsharp/fslang-suggestions/issues/1038), [PR #14238](https://github.com/dotnet/fsharp/pull/14238))

    > **If you believe that release notes are not necessary for this PR, please add `NO_RELEASE_NOTES` label to the pull request.**

### CODE REVIEW

**Review by vzarytovskii — APPROVED** [review]

File: tests/benchmarks/FCSBenchmarks/CompilerServiceBenchmarks/CE100xnest10.fs
Comment (psfinaki) [inline tests/benchmarks/FCSBenchmarks/CompilerServiceBenchmarks/CE100xnest10.fs:1]: I must say - scrolling through this file is a somehow soothing process.

File: tests/benchmarks/FCSBenchmarks/CompilerServiceBenchmarks/CEwCO500xnest1.fs
Comment (psfinaki) [inline tests/benchmarks/FCSBenchmarks/CompilerServiceBenchmarks/CEwCO500xnest1.fs:1]: Small nitpick - probably put these files into a special folder to have a better structured project.

**Review by psfinaki — APPROVED** [review]
Great stuff - thanks! 

I am currently working on doing some CI checks for benchmarks, as of now there is at least a build of the bench solution, we will probably also have some smoke test execution of the tests to be more sure the code stays up to date.

File: tests/benchmarks/FCSBenchmarks/CompilerServiceBenchmarks/CEwCO500xnest1.fs
Comment (psfinaki) [reply tests/benchmarks/FCSBenchmarks/CompilerServiceBenchmarks/CEwCO500xnest1.fs:1]: Thanks!

**Pull request merged**